Sources/Controllers/DashboardOnMap/MapSettings/OAMapSettingsMainScreen.mm

```diff
@@ -224,170 +233,152 @@ - (void) setupView
-                @"groupName": OALocalizedString(@"map_settings_style"),
+                @"group_name": OALocalizedString(@"map_settings_style"),
                 @"cells": mapStyleSectionData
         }];
     }
 
     NSMutableArray *overlayUnderlaySectionData = [NSMutableArray array];
-
-    if ([_iapHelper.srtm isActive])
-        [overlayUnderlaySectionData addObject:@{
-                @"name": OALocalizedString(@"shared_string_terrain"),
-                @"description": @"",
-                @"secondaryImg": @"ic_action_additional_option",
-                @"type": [OASettingSwitchCell getCellIdentifier],
-                @"key": @"terrain_layer"
-        }];
-
+    [overlayUnderlaySectionData addObject:@{
+            @"name": OALocalizedString(@"shared_string_terrain"),
+            @"image": hasSRTM ? @"ic_custom_hillshade" : @"ic_custom_contour_lines_colored",
+            hasSRTM ? @"options" : @"desc": hasSRTM ? @YES : OALocalizedString(@"contour_lines_hillshades_slope"),
+            @"type": hasSRTM ? [OAIconTextDividerSwitchCell getCellIdentifier] : [OAPromoButtonCell getCellIdentifier],
+            @"key": @"terrain_layer"
+    }];
     [overlayUnderlaySectionData addObject:@{
             @"name": OALocalizedString(@"map_settings_over"),
-            @"description": @"",
-            @"secondaryImg": @"ic_action_additional_option",
-            @"type": [OASettingSwitchCell getCellIdentifier],
+            @"image": @"ic_custom_overlay_map",
+            @"options": @YES,
+            @"type": [OAIconTextDividerSwitchCell getCellIdentifier],
             @"key": @"overlay_layer"
     }];
     [overlayUnderlaySectionData addObject:@{
             @"name": OALocalizedString(@"map_settings_under"),
-            @"description": @"",
-            @"secondaryImg": @"ic_action_additional_option",
-            @"type": [OASettingSwitchCell getCellIdentifier],
+            @"image": @"ic_custom_underlay_map",
+            @"options": @YES,
+            @"type": [OAIconTextDividerSwitchCell getCellIdentifier],
             @"key": @"underlay_layer"
     }];
 
     [data addObject:@{
-            @"groupName": OALocalizedString(@"map_settings_overunder"),
+            @"group_name": OALocalizedString(@"map_settings_overunder"),
             @"cells": overlayUnderlaySectionData
     }];
 
     [data addObject:@{
-            @"groupName": OALocalizedString(@"language"),
+            @"group_name": OALocalizedString(@"language"),
             @"cells": @[@{
                     @"name": OALocalizedString(@"sett_lang"),
                     @"value": [self getMapLangValueStr],
-                    @"type": [OASettingsTableViewCell getCellIdentifier],
+                    @"image": @"ic_custom_map_languge",
+                    @"type": [OAIconTitleValueCell getCellIdentifier],
                     @"key": @"map_language"
             }]
     }];
 
     tableData = data;
-    [tblView reloadData];
+    [UIView transitionWithView: tblView
```

Comment:
I'm not sure the reload is needed in this method at all. Secondly, do not use the animated reload if it's not even visible to the user.